src/Service.Tests/SqlTests/GraphQLQueryTests/GraphQLQueryTestBase.cs

```diff
@@ -387,6 +387,228 @@ public virtual async Task MultipleResultJoinQuery()
+
+            string expected = @"
+                {
+                  ""__typename"": ""StockConnection""
+                }
+            ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLQuery, graphQLQueryName, isAuthenticated: true);
+            SqlTestHelper.PerformTestEqualJsonStrings(expected, actual.ToString());
+        }
+
+        /// <summary>
+        /// Validates that a point query with only __typename field in the selection set
+        /// returns the right type
+        /// </summary>
+        [TestMethod]
+        public async Task PointQueryWithTypenameInSelectionSet()
+        {
+            string graphQLQueryName = "book_by_pk";
+            string graphQLQuery = @"{
+                book_by_pk(id: 3) {
+                    __typename
+                }
+            }";
+
+            string expected = @"
+                {
+                    ""__typename"": ""book""
+                }
+            ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLQuery, graphQLQueryName, isAuthenticated: false);
+            SqlTestHelper.PerformTestEqualJsonStrings(expected, actual.ToString());
+        }
+
+        /// <summary>
+        /// Validates that a nested point query with only __typename field in each selection set
+        /// returns the right types
+        /// </summary>
+        [TestMethod]
+        public async Task NestedPointQueryWithOnlyTypenameInEachSelectionSet()
+        {
+            string graphQLQueryName = "book_by_pk";
+            string graphQLQuery = @"{
+                book_by_pk(id: 3) {
+                    __typename
+                    publishers {
+                      __typename
+                      books {
+                        __typename
+                      }
+                    }
+                }     
+            }";
+
+            string expected = @"
+                {
+                  ""__typename"": ""book"",
+                  ""publishers"": {
+                    ""__typename"": ""Publisher"",
+                    ""books"": {
+                      ""__typename"": ""bookConnection""
+                    }
+                  }
+                }
+            ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLQuery, graphQLQueryName, isAuthenticated: false);
+            SqlTestHelper.PerformTestEqualJsonStrings(expected, actual.ToString());
+        }
+
+        /// <summary>
+        /// Validates that querying a SP with only __typename field in the selection set
+        /// returns the right type(s)
+        /// </summary>
+        public virtual async Task QueryAgainstSPWithOnlyTypenameInSelectionSet()
+        {
+            string graphQLQueryName = "executeGetBooks";
+            string graphQLQuery = @"{
+                executeGetBooks{
+                    __typename
+                }     
+            }";
+
+            string expected = @"
+                [
+      {
```

Comment:
What decides how many __typename items should be returned back as part of the response array?

Reply:
There would be no change in the number of items returned in the response when the selection set contains entity fields vs just __typename field. 

- For instance, let's say a policy is defined - `@item.id ne 1`. This would be honored and the response will not contain an entry `__typename: <actual_type>` for id corresponding to 1. 

- For a query such as this, 
```graphql
{
  book_by_pk (id: 11111){
    __typename
  }

}

```
where id `11111` is not present in the table, the response will be the following:

```graphql
{
  "data": {
    "book_by_pk": null
  }
}
```
which is the same when the selection set contains entity fields such as `id`, `title`, etc.



Essentially, the number of items in the response array will be the same except that it would contain just the type names.